### PR TITLE
calc: specify license

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Follow-up #113890

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tomberek** [comment]
We can conditional the license on the bool. Eg: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/logstash/6.x.nix#L51

**Comment by Mic92** [comment]
> We can conditional the license on the bool. Eg: [https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/logstash/6.x.nix#L51](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/misc/logstash/6.x.nix?rgh-link-date=2021-02-22T16%3A13%3A47Z#L51)

both licenses are free, so I don't think we need this here.

File: pkgs/applications/science/math/calc/default.nix [hunk 1]
```diff
@@ -41,7 +41,9 @@ stdenv.mkDerivation rec {
   meta = with lib; {
     description = "C-style arbitrary precision calculator";
     homepage = "http://www.isthe.com/chongo/tech/comp/calc/";
-    license = licenses.lgpl21Only;
+    # The licensing situation depends on readline (see section 3 of the LGPL)
+    # If linked against readline then GPLv2 otherwise LGPLv2.1
+    license = with licenses; [ gpl2Only lgpl21Only ];
```
Comment (Mic92) [inline pkgs/applications/science/math/calc/default.nix:46]:
```suggestion
    license = with licenses; if enableReadline then gpl2Only else lgpl21Only;
```

Comment (fabaff) [reply pkgs/applications/science/math/calc/default.nix:46]: This make much more sense. Thanks.

**Pull request merged**